napalm/ios/ios.py

```diff
@@ -436,13 +466,16 @@ def commit_config(self):
         output += self.device.send_command_expect("write mem")
 
     def discard_config(self):
+        """Discard loaded candidate configurations."""
+        self._discard_config()
+
+    @_file_operation
+    def _discard_config(self):
```

Comment:
Why did you need to make a shadow method here (as opposed to just decorating `discard_config`?

Reply:
This was an interesting py2 vs py3 issue that I hadn't seen before:
In 2.7, `functools.wraps()` doesn't fix the signature of `wrapper()` to match the original function.
As a result, `test.ios.test_getters.TestGetter.test_method_signatures()` fails. Since that test skips private methods the easiest way to work-around was to decorate a private method and call it from the public one.
If you're aware of a more elegant fix then I'm happy to incorporate that?

Reply:
Okay, makes sense. No I am not aware of a better solution...so we should just use your solution.